### DISCUSSION TITLE
Serial output weird behavior

### DISCUSSION THREAD

**Discussion by BreadJS** [nomis/logitech-z906#9] [Q&A]
Hello,

First of, thank you very much for the documentation! It really helped me! Although, when sending the status request I get some verry weird data back thats starts with `3F` and the length of hex will change on almost every output. Currently I'm in bed writing this and can show you some full outputs tomorrow! It would really mean a lot if you could help me with this issue as I'm writing a ESPhome template to use the speakers with home assistant! I will be crediting your work btw!

Thanks in advance and have a early weekend!

**Comment by BreadJS** [comment]
I saw a few weeks ago that the new one is missing bytes. Code length was 23 but mine was 24. I'm not 100% sure what is missing, but I will be comparing the 2 and make PR for the fix. Currently doing lots of things at once so it's a bit unorganized for me.

**Reply by nomis** [reply]
You'll need to provide a copy of the messages being sent and received otherwise I can't really comment on it.

I have raised the difference in length between https://github.com/zarpli/Logitech-Z906/issues/4 (based on the decompiled firmware from the amplifier) and my observation of the communication between devices but that issue was closed without explanation.
